ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryLedgerReaderWriter.scala

```diff
@@ -46,38 +40,59 @@ private[memory] class InMemoryState(
     val state: mutable.Map[ByteString, Array[Byte]] = mutable.Map.empty,
 )
 
+// Dispatcher and LedgerStateAccess are passed in to allow for a multi-participant setup.
 final class InMemoryLedgerReaderWriter(
     override val ledgerId: LedgerId,
     override val participantId: ParticipantId,
     now: () => Instant,
     dispatcher: Dispatcher[Index],
+    ledgerStateAccess: InMemoryLedgerStateAccess,
```

Comment:
Win! But why not inject `InMemoryState` rather than `InMemoryLedgerStateAccess`, and move the lock into that object? `InMemoryLedgerStateAccess` doesn't _feel_ like a public API.

Reply:
Fully agreed -- that's how it's done in the integration kit as well.

Reply:
Thanks, that's a great suggestion.